app/controllers/reports/regions_controller.rb

```diff
@@ -117,7 +117,8 @@ def monthly_district_data_report
         raise ActiveRecord::RecordNotFound, "unknown report_scope #{report_scope}"
       end
     }
-    @period = Period.month(Date.current)
+    period_end = params[:period_end] || Date.current
+    @period = Period.month(period_end)
```

Comment:
This can just be something like this:

```
@current_period = Period.month(params[:period] || Date.current)
```

The fact that it is the end of the range of data is implicit through all these controlled actions.